baybe/utils/botorch_wrapper.py

```diff
@@ -3,6 +3,8 @@
 import torch
 from botorch.test_functions import SyntheticTestFunction
 
+from baybe.utils.torch import DTypeFloatTorch
```

Comment:
Do we require any tests for this? I verifies locally that both `x_tensor` and the `result` have `torch.float32` precision.

Reply:
I think that suffices for now. FYI: remember that the wrapper is legacy code anyway, i.e. removing it and refactoring the callable interface in the simulation module (together with lots of other stuff there) is overdue since long. Hence, all good from my end 👍🏼

Reply:
Waiting then for @Scienfitz to approve.

Reply:
did I miss something? I approved this. any other response required?

Reply:
Nope, everything fine. Will rebase and merge.